src/xma/include/lib/xma_buffer.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XMA_BUFFERCLS_H_
+#define _XMA_BUFFERCLS_H_
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <atomic>
+#include "xrt.h"
+#include "core/include/experimental/xrt_bo.h"
+
+namespace xma_core {
+namespace plg {
+
+using memory_group = xrtMemoryGroup;
+
+/**
+ * class buffer : abstraction for xma buffer
+ */
+class buffer: public xrt::bo
```

Comment:
No, I don't think this is what you want to do.
It may or may not make sense to have your own buffer class that contains an xrt::bo object, but you definitely don't want to derived from the class.

Reply:
It is incorrect to blindly disallow derived classes. In this case it is fine to use xrt::bo as a member. Changed.